torch/distributed/optim/named_optimizer.py

```diff
@@ -284,6 +284,19 @@ def add_param_group(self, param_group: Mapping[str, Any]) -> None:
         # Update param_groups from optimizer.
         self.param_groups = self._optimizer.param_groups
 
+    def init_state(self) -> None:
+        """
+        Runs a dummy optimizer step, which allows to initialize optimizer state
+        because we do lazy init for most optimizers.
+
+        This allows doing in-place loading of optimizer state from a checkpoint.
+        """
+        for _, param in self.named_parameters.items():
+            if param.requires_grad:
+                t = torch.zeros_like(param)
+                param.grad = torch.autograd.Variable(t)
+        self.step(closure=None)
```

Comment:
nit: add a comment mentioning this is what loads the initial state for optimizer states?